lian/src/lian_nav2/src/lian_planner.cpp

```diff
@@ -20,14 +23,45 @@ void LianPlaner::configure(
   tf_ = tf;
   planner_->costmap_ = costmap_ros->getCostmap();
   global_frame_ = costmap_ros->getGlobalFrameID();
-
+  
 }
 
-nav_msgs::msg::Path ThetaStarPlanner::createPlan(
+nav_msgs::msg::Path LianPlaner::createPlan(
   const geometry_msgs::msg::PoseStamped & start,
   const geometry_msgs::msg::PoseStamped & goal)
 {
-  nav_msgs::msg::Path global_path;
-  auto start_time = std::chrono::steady_clock::now();
   
+  searchParams[CNS_TAG_SX] = start.pose.position.x
+  searchParams[CNS_TAG_SY] = start.pose.position.y
+  searchParams[CNS_TAG_FX] = goal.pose.position.x
+  searchParams[CNS_TAG_FY] = goal.pose.position.y
+
+  nav_msgs::msg::Path lian_path;
+  auto start_time = std::chrono::steady_clock::now();
+
+  if (sr.pathfound) {
+    lian_path.header.stamp = node_->now();
+    lian_path.header.frame_id = map;
+
+    for(int node=0; node<len(sr.hppath);node++){
+
+      geometry_msgs::msg::PoseStamped posestamped;
+
+      tf2::Quaternion nodeQuaternion;
+      nodeQuaternion.setRPY(0,0,sr.hppath[node].angle);
+
+      posestamped.pose.position.x = sr.hppath[node].i;
+      posestamped.pose.position.y = sr.hppath[node].j;
+      posestamped.pose.orientation.z = nodeQuaternion.getZ();
```

Comment:
Bonus question, performance wise, would it be better to init a local variable that will hold `sr.hppath[node]`, instead of calling it 3 times?